roles/collect/tasks/main.yml

```diff
@@ -113,8 +120,23 @@
 - name: append filename to fact list
   set_fact:
     csv_files: "{{ csv_files + [item.dest | basename] }}"
+    csv_files_full_path: "{{ csv_files_full_path + [item.dest] }}"
   with_items: "{{ download_result.results }}"
 
+- name: Check that required files exist
+  stat:
+    path: '{{ item }}'
+  register: csv_stat_result
+  with_items:
+    - '{{ csv_files_full_path }}'
```

Comment:
It isn't necessary to create a duplicate fact. You should be able to reuse the `csv_files` fact in the stat with something like `{{ csv_path + item }}"`